CMakeLists.txt

```diff
@@ -20,17 +20,46 @@ else ()
     message(STATUS "Building with 1.10 CMake scripts.")
     string(CONCAT DESCRIPTION_STRING "The AWS SDK for C++ provides a modern C++ (standard version C++11 or later) "
             "interface for Amazon Web Services (AWS).")
+
+    if (NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+        message(STATUS "Setting build type to 'RelWithDebInfo' as none was specified.")
+        set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE
+                STRING "Choose the type of build." FORCE)
+        # cmake-gui helper
+        set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
```

Comment:
This helps gui tools as cmake to show only a list of valid options to configure the CMAKE_BUILD_TYPE variable in place of having a free form string as input.